prowler/providers/aws/services/cognito/cognito_service.py

```diff
@@ -100,13 +154,180 @@ def __get_user_pool_mfa_config__(self):
+                    self.identity_pools[identity_pool.arn].associated_pools = (
+                        identity_pool_details.get("CognitoIdentityProviders", [])
+                    )
+                    self.identity_pools[identity_pool.arn].tags = (
+                        identity_pool_details.get("IdentityPoolTags", {})
+                    )
+                    self.identity_pools[
+                        identity_pool.arn
+                    ].allow_unauthenticated_identities = identity_pool_details.get(
+                        "AllowUnauthenticatedIdentities", False
+                    )
+                except Exception as error:
+                    logger.error(
+                        f"{identity_pool.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                    )
+        except Exception as error:
+            logger.error(
+                f"{identity_pool.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def __get_identity_pool_roles__(self):
+        logger.info("Cognito - Getting Identity Pool Roles...")
+        try:
+            for identity_pool in self.identity_pools.values():
+                try:
+                    identity_pool_roles = self.regional_clients[
+                        identity_pool.region
+                    ].get_identity_pool_roles(IdentityPoolId=identity_pool.id)
+                    self.identity_pools[identity_pool.arn].roles = IdentityPoolRoles(
+                        authenticated=identity_pool_roles.get("Roles", {}).get(
+                            "authenticated", ""
+                        ),
+                        unauthenticated=identity_pool_roles.get("Roles", {}).get(
+                            "unauthenticated", ""
+                        ),
+                    )
+                except Exception as error:
+                    logger.error(
+                        f"{identity_pool.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                    )
+        except Exception as error:
+            logger.error(
+                f"{identity_pool.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
 
 class MFAConfig(BaseModel):
     sms_authentication: Optional[dict]
     software_token_mfa_authentication: Optional[dict]
     status: str
 
 
+class AccountTakeoverRiskConfiguration(BaseModel):
+    low_action: Optional[dict]
+    medium_action: Optional[dict]
+    high_action: Optional[dict]
+
+
+class CompromisedCredentialsRiskConfiguration(BaseModel):
+    event_filter: Optional[list]
+    actions: Optional[dict]
+
+
+class RiskConfiguration(BaseModel):
+    compromised_credentials_risk_configuration: Optional[
+        CompromisedCredentialsRiskConfiguration
+    ]
+    account_takeover_risk_configuration: Optional[AccountTakeoverRiskConfiguration]
+
+
+class UserPoolClient(BaseModel):
```

Comment:
Please add the ID and Name of the client when doing the list

Reply:
Done